app/core/SanitizationMiddleware.ts

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+import { JsonRpcMiddleware, JsonRpcRequest } from 'json-rpc-engine';
+import { addHexPrefix } from 'ethereumjs-util';
+
+interface TransactionParamsLike {
+  from: string;
+  to: string;
+  value: string;
+  data: string;
+  gas: string;
+  gasPrice: string;
+  nonce: string;
+  fromBlock: string;
+  toBlock: string;
+  address: string;
+  topics: string[];
+  [k: string]: string | string[];
+}
+
+// we use this to clean any custom params from the txParams
+const permittedKeys = [
+  'from',
+  'to',
+  'value',
+  'data',
+  'gas',
+  'gasPrice',
+  'nonce',
+  'fromBlock',
+  'toBlock',
+  'address',
+  'topics',
+];
+
+function cloneTxParams(txParams: TransactionParamsLike | any) {
+  const sanitized = permittedKeys.reduce(
+    (copy, permitted: keyof TransactionParamsLike) => {
+      if (permitted in txParams) {
+        if (Array.isArray(txParams[permitted])) {
+          copy[permitted] = (txParams[permitted] as string[]).map((item) =>
+            sanitize(item),
+          );
+        } else {
+          copy[permitted] = sanitize(txParams[permitted]);
+        }
+      }
+      return copy;
+    },
+    {} as TransactionParamsLike,
+  );
+
+  return sanitized;
+}
+
+function sanitize(value: any) {
```

Comment:
No need to disable type checking here:
```suggestion
function sanitize(value: unknown) {
```